src/stores/advertises.js

```diff
@@ -78,6 +95,38 @@ export const useAdvertiseStore = defineStore('advertises', {
         }
       }
     },
+    async fetchAdvertiseById(campaignId) {
+      let advertise
+      if (this.getALlActiveAdvertises.length || this.getAdvertises.length) {
+        advertise =
+          this.getALlActiveAdvertises?.find((advertise) => advertise.id === campaignId) ||
+          this.getAdvertises?.find((advertise) => advertise.id === campaignId)
+      }
+      if (advertise) {
+        return advertise
+      }
+      try {
+        this.setLoaderTrue()
+        const userStore = useUserStore()
+        const docData = await getDocs(query(collection(db, 'advertises'), where('id', '==', campaignId)))
+        if (docData.empty) {
+          return await this.redirect()
+        }
+        advertise = docData.docs[0].data()
+        if (advertise.author.id === userStore.getUserId) {
+          advertise.author = userStore.getUser
+        } else if (userStore.isAdmin) {
+          advertise.author = userStore.getUserById(advertise.author.id)
+        } else {
+          return await this.redirect()
+        }
+        return advertise
+      } catch (error) {
+        return await this.redirect()
```

Comment:
And we should handle the error somehow to know what is the issue. Add a console.error or add it to error store (or both)